src/IK.h

```diff
@@ -0,0 +1,69 @@
+// This file is part of RBDyn.
+//
+// RBDyn is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// RBDyn is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with RBDyn.  If not, see <http://www.gnu.org/licenses/>.
+
+#pragma once
+
+// includes
+// std
+#include <vector>
+
+// SpaceVecAlg
+#include <SpaceVecAlg/SpaceVecAlg>
+
+#include "Jacobian.h"
+
+namespace rbd
+{
+class MultiBody;
+class MultiBodyConfig;
+
+/**
+	* Inverse Kinematics algorithm.
+	*/
+class InverseKinematics
+{
+public:
+	InverseKinematics()
+	{}
+	/// @param mb MultiBody associated with this algorithm.
+	InverseKinematics(const MultiBody& mb, int ef_index);
+	/**
+	* Compute the inverse kinematics.
+	* @param mb MultiBody used has model.
+	* @param mbc Use q generalized position vector
+	* @return bool if computation has converged
+	* Fill q with new generalized position
```

Comment:
I think that anyway the user expects that the `MultiBodyConfig` passed to the algorithm will be modified so this kind of back-tracking should be left to the user. Furthermore, it may be of some interest to get the partial result. It should however be specified in the documentation.